docs/docs/features/endpoints.md

```diff
@@ -6,7 +6,7 @@ title: Endpoints
 OAuth2 Proxy responds directly to the following endpoints. All other endpoints will be proxied upstream when authenticated. The `/oauth2` prefix can be changed with the `--proxy-prefix` config variable.
 
 - /robots.txt - returns a 200 OK response that disallows all User-agents from all paths; see [robotstxt.org](http://www.robotstxt.org/) for more info
-- /ping - returns a 200 OK response, which is intended for use with health checks
+- /ping - returns a 200 OK response, which is intended for use with health checks. If the `X-Healthcheck: full` header is present, it performs a deeper check for underlying connections (e.g., Redis store)
```

Comment:
Would it be better to have a separate endpoint? Can for example a Kubernetes readiness check configure headers to send?

Reply:
I started with a different path and converted to using the headers instead.
I had the _manifests_ in mind (https://github.com/oauth2-proxy/manifests/pull/113) and was looking for a suitable upgrade strategy (how people won't get confused and deploy a new manifest with an old version that doesn't support the new path).

That said, I have no hard feelings about using headers vs. different paths (e.g., `/ping` and `/ready`). Maybe I could use [Helm's `semverCompare`](https://helm.sh/docs/chart_template_guide/function_list/#semvercompare) to make this distinction instead.

What do you think?

Reply:
Is the version of the helm chart not tied to the version of the application within it? My understanding is that if we add the new flags to the helm chart, they will only be done so once a new version of OAuth2 proxy is released

I would lean towards a separate endpoint as I think in general that has more compatibility